### PR TITLE
Log errors to Google Analytics

### PR DESCRIPTION
**Pull request by artpi**
In #6360 with @scruffian we introduced error logging mechanism that logs errors to our REST-API endpoint.
That is all dandy, we have plenty errors in our IRC channel, but we would like so much more context!
Well, there is a place we have plenty of context for current user already:

#### Google analytics

This PR sends errors from 1% of our users to Google Analytics, so we can slice & dice and extract valuable data.

## Testing 

Because GA script works only for non-proxied `wordpress.com`, this is not going to be easy.

- Grab credentials to our `Atlas&Calypso` GA property
- Un-proxy yourself ( serious stuff! )
- restart browser so you are really unproxied
- Remove hardcoded protection in `server/pages/index.jade` (to test you need to remove the line "development") :
```
		if 'development' !== env
			script( src=urls['catch-js-errors.js'] )
```
- break something [ call potato() in some render for example ]
- build project (after removing the line - the live reload does not work for jade)
- Put yourself in logging group: `localStorage.setItem('log-errors', 'analytics');`
- hard reload
- Validate that you have onerror script `window.onerror`
- Spoof GA call in your console: `window.ga = ( ...args ) => console.log( JSON.stringify( args ) );`
- navigate where you broke stuff
- see error on console. Copy the content without `[]`.
- Navigate to http://wordpress.com
- Call GA script with copied data. It should look like: 
```
window.ga("send","event","JS Error","Uncaught ReferenceError: potato is not defined","URL: http://calypso.localhost:3000/stats/insights/artpitesting17.wordpress.com| http://calypso.localhost:3000/stats/insights/artpitesting17.wordpress.com:26",0,{"nonInteraction":true});
```
- [ENTER]
- Open GA `Real Time` -> `Events` -> `Events (Last 30 min)`
- Search `JS Error` on the search box
- See your error.
- Profit.

zrzut ekranu 2016-07-01 o 17 03 16

CC @gwwar @rralian @scruffian 

Test live: https://calypso.live/?branch=update/errors-analytics

### CODE REVIEW

File: public/catch-js-errors.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ( function() {
 	var savedWindowOnError = window.onerror,
 		assignment = 1,
+		localStorageAssignment = 'false',
```
Comment (gwwar) [inline public/catch-js-errors.js]: Maybe `errorLogger`? The fact that we stored this in localStorage isn't super important.

File: public/catch-js-errors.js [hunk 2]
```diff
@@ -108,8 +111,8 @@
 			}
 		}
 
-		if ( localStorage.getItem( 'log-errors' ) !== undefined &&
-			( localStorage.getItem( 'log-errors' ) === 'rest-api' || localStorage.getItem( 'log-errors' ) === 'true' )
+		if ( localStorageAssignment !== undefined &&
+			( localStorageAssignment === 'rest-api' || localStorageAssignment === 'true' )
```
Comment (gwwar) [inline public/catch-js-errors.js]: Maybe we could fix any localStorageAssignments that are set to 'true' before this if block.

File: public/catch-js-errors.js [hunk 3]
```diff
@@ -108,8 +111,8 @@
 			}
 		}
 
-		if ( localStorage.getItem( 'log-errors' ) !== undefined &&
-			( localStorage.getItem( 'log-errors' ) === 'rest-api' || localStorage.getItem( 'log-errors' ) === 'true' )
+		if ( localStorageAssignment !== undefined &&
```
Comment (gwwar) [inline public/catch-js-errors.js]: Why do we need to check for undefined? We're doing a equality check after this that shouldn't throw any errors.

File: public/catch-js-errors.js [hunk 4]
```diff
@@ -118,6 +121,23 @@
 					savedWindowOnError();
 				}
 			};
+		} else if ( localStorageAssignment === 'analytics' ) {
```
Comment (gwwar) [inline public/catch-js-errors.js]: Maybe the assignment should be 'ga'? Would we never log to other analytics tools like tracks?

File: public/catch-js-errors.js [hunk 5]
```diff
@@ -118,6 +121,23 @@
...
+			window.onerror = function( message, scriptUrl, lineNumber ) {
+				if ( typeof window.ga === 'function' ) {
+					window.ga(
```
Comment (gwwar) [inline public/catch-js-errors.js]: Related documentation for ga events https://developers.google.com/analytics/devguides/collection/analyticsjs/events#overview

File: public/catch-js-errors.js [hunk 6]
```diff
@@ -118,6 +121,23 @@
...
+						'send',
+						'event',
+						'JS Error',
+						message,
+						'URL: ' + document.location.href + '| ' + scriptUrl + ':' + lineNumber,
+						0,
```
Comment (gwwar) [inline public/catch-js-errors.js]: 👍 The default eventValue is also `0` but it doesn't hurt to add it here.

**Comment by gwwar** [comment]
👍 I took this for a spin. I left a few notes, but I think this is good to 🚢

File: public/catch-js-errors.js (hunk 4 above)
Comment (artpi) [inline public/catch-js-errors.js]: Too late :/ That key is already deployed #6360 and is in everybody's LocalStorage :/

Comment (gwwar) [inline public/catch-js-errors.js]: Ah gotcha 👍 , this wasn't a blocker anyway.

File: public/catch-js-errors.js (hunk 1 above)
Comment (artpi) [inline public/catch-js-errors.js]: Done!

File: public/catch-js-errors.js (hunk 2 above)
Comment (artpi) [inline public/catch-js-errors.js]: Done!

File: public/catch-js-errors.js (hunk 3 above)
Comment (artpi) [inline public/catch-js-errors.js]: Done!

**Pull request merged**